### ISSUE TITLE
Enum field type doesn't allow for non-space whitespace in declaration

### ISSUE THREAD

**Issue by mcccclean** [silverstripe/silverstripe-framework#4683]
Declaring a long enum over multiple lines will not work correctly.

For example:
```php
public static $db = array(
    $fruit => 'Enum("
        apple,
        banana,
        etc
        ")'
);
```

It won't cause any errors in `dev/build` but will just result in rejecting seemingly-fine values invalid as they have mismatched whitespace.

The fix is simple (I've done it locally and it works fine) `Enum::__construct` should split the input string according to `/\s*,\s*/` instead of `/ *, */`.

**Comment by kinglozzer** [comment]
@mcccclean You should be able to do the following:

```php
public static $db = array(
    $fruit => 'Enum(array(
        "apple",
        "banana",
        "etc"
    ))'
);
```

Given that we do support passing a single string arg that’s split on commas, I don’t see any harm in this change as it doesn’t complicate the code at all.

**Comment by dhensby** [comment]
Ah yes, this has annoyed me before too - was just too lazy to fix it. Fix now PRed

**Comment by tractorcow** [comment]
Fixed with #4684
